### ISSUE TITLE
Using RedundantInitialize makes RuboCop warn about RedundantCopDisableDirective

### ISSUE THREAD

**Issue by cyrorama** [rubocop/rubocop#11025]
## Expected behavior

I expect using `# rubocop:disable Style/RedundantInitialize` to prevent RuboCop errors.

## Actual behavior

Using `# rubocop:disable Style/RedundantInitialize` results in warnings about `Lint/RedundantCopDisableDirective` when the disable is actually needed.

Here is an example:
```ruby
        Class.new do
          def initialize(foo:, bar:) end # rubocop:disable Style/RedundantInitialize
        end
```

This results in the following:
```
spec/file.rb:315:42: W: [Correctable] Lint/RedundantCopDisableDirective: Unnecessary disabling of Style/RedundantInitialize.
          def initialize(foo:, bar:) end # rubocop:disable Style/RedundantInitialize
                                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Adding in a disabling of `Lint/RedundantCopDisableDirective` results in no issues:
```ruby
        Class.new do
          def initialize(foo:, bar:) end # rubocop:disable Style/RedundantInitialize, Lint/RedundantCopDisableDirective
        end
```

Here's the result: 
```
4932 files inspected, no offenses detected
```

And if I remove any disabling, I get the error about `RedundantInitialize`:
```ruby
        Class.new do
          def initialize(foo:, bar:) end
        end
```

```
spec/file.rb:315:11: C: [Correctable] Style/RedundantInitialize: Remove unnecessary empty initialize method.
          def initialize(foo:, bar:) end
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

## Steps to reproduce the problem

Have an empty initialize within a class, and place a `# rubocop:disable Style/RedundantInitialize`. 

The example I posted is from a spec file, but I've found this to be true with other parts of our codebase outside of specs.

Ex:
```ruby
  class Example
    def initialize(foo:, bar:); end # rubocop:disable Style/RedundantInitialize
```

Output:
```
lib/example.rb:5:37: W: [Correctable] Lint/RedundantCopDisableDirective: Unnecessary disabling of Style/RedundantInitialize.
    def initialize(foo:, bar:); end # rubocop:disable Style/RedundantInitialize
                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

But this works:
```ruby
  class Example
    def initialize(foo:, bar:); end # rubocop:disable Style/RedundantInitialize, Lint/RedundantCopDisableDirective
```

Output:
```
4932 files inspected, no offenses detected
```

## RuboCop version

```
╰─± BUNDLE_GEMFILE=.overcommit/Gemfile bundle exec rubocop -V
1.36.0 (using Parser 3.1.2.1, rubocop-ast 1.21.0, running on ruby 2.7.6) [x86_64-darwin20]
  - rubocop-performance 1.13.3
  - rubocop-rails 2.14.2
  - rubocop-rspec 2.9.0
```

**Comment by LucasCarrias** [comment]
May I take this issue?
